core/src/Revolution/modResource.php

```diff
@@ -1534,4 +1536,81 @@ public function clearCache($context = '')
             $this->xpdo->invokeEvent('OnResourceCacheUpdate', ['id' => $this->get('id')]);
         }
     }
+
+    /**
+     * Get icon classes for the resource, based on template, class_key, system settings, etc.
+     *
+     * @return array
+     */
+    public function getIconClasses()
+    {
+        try {
+            $reflectionClass = new ReflectionClass($this->get('class_key'));
+            $classKey = strtolower($reflectionClass->getShortName());
+        } catch (ReflectionException $e) {
+            $classKey = strtolower($this->get('class_key'));
+        }
+
+        if (substr($classKey, 0, 3) === 'mod') {
+            $classKey = substr($classKey, 3);
+        }
+
+        $iconCls = [];
+
+        $contentType = $this->getOne('ContentType');
+        if ($contentType && $contentType->get('icon')) {
+            $iconCls[] = $contentType->get('icon');
+        }
+
+        $template = $this->getOne('Template');
+        $tplIcon = '';
+        if ($template && $template->get('icon')) {
+            $tplIcon = $template->get('icon');
+            $iconCls[] = $template->get('icon');
+        }
+
+        $classKeyIcon = $this->xpdo->getOption('mgr_tree_icon_' . $classKey, null, 'tree-resource', true);
+        if (empty($iconCls)) {
+            $iconCls[] = $classKeyIcon;
+        }
+
+        switch ($classKey) {
+            case 'weblink':
+                $iconCls[] = $this->xpdo->getOption('mgr_tree_icon_weblink', null, 'tree-weblink');
+                break;
+
+            case 'symlink':
+                $iconCls[] = $this->xpdo->getOption('mgr_tree_icon_symlink', null, 'tree-symlink');
+                break;
+
+            case 'staticresource':
+                $iconCls[] = $this->xpdo->getOption('mgr_tree_icon_staticresource', null, 'tree-static-resource');
+                break;
+        }
+
+        // Icons specific with the context and resource ID for super specific tweaks
+        $iconCls[] = 'icon-' . $this->get('context_key') . '-' . $this->get('id');
+        $iconCls[] = 'icon-parent-' . $this->get('context_key') . '-' . $this->get('parent');
+
+        // Modifiers to indicate resource _state_
+        $childrenCount = $this->xpdo->getCount(modResource::class, ['parent' => $this->get('id')]);
+        if ($childrenCount > 0 || $this->isfolder) {
+            if (empty($tplIcon) && $classKeyIcon === 'tree-resource') {
+                $iconCls[] = $this->xpdo->getOption('mgr_tree_icon_folder', null, 'parent-resource');
+            }
+        }
+
+        return $iconCls;
+    }
+
+    public function getStatusClasses()
+    {
+        $classes = [];
+
+        if (!$this->get('published')) $classes[] = 'unpublished';
+        if ($this->get('deleted')) $classes[] = 'deleted';
+        if ($this->get('hidemenu')) $classes[] = 'hidemenu';
```

Comment:
A formatting nitpick, but we should conform to our own PSR rules:
```suggestion
        if (!$this->get('published')) {
            $classes[] = 'unpublished';
        }
        if ($this->get('deleted')) {
            $classes[] = 'deleted';
        }
        if ($this->get('hidemenu')) {
            $classes[] = 'hidemenu';
        }
```